File: src/1464.maximum_product_of_two_elements_in_an_array.kt
fun main() {
    println(maxProduct(intArrayOf(3, 4, 5, 2)))
    println(maxProduct(intArrayOf(1, 5, 4, 5)))
    println(maxProduct(intArrayOf(3, 7)))
}

fun maxProduct(nums: IntArray): Int {
//    var length = nums.size - 1
//
//    while (length > 0) {
//        for (i in 0..<length) {
//            val left = nums[i]
//            val right = nums[i + 1]
//
//            if (left > right) {
//                nums[i] = right
//                nums[i + 1] = left
//            }
//        }
//
//        length--
//    }

    nums.sort()

    return (nums.last() - 1) * (nums[nums.size - 2] - 1)
}

File: lib/1464.maximum_product_of_two_elements_in_an_array.dart
void main() {
  print(maxProduct([3, 4, 5, 2]));
  print(maxProduct([1, 5, 4, 5]));
  print(maxProduct([3, 7]));
}

int maxProduct(List<int> nums) {
  var length = nums.length - 1;

  while (length > 0) {
    for (var i = 0; i < length; i++) {
      final left = nums[i];
      final right = nums[i + 1];

      if (left > right) {
        nums[i] = right;
        nums[i + 1] = left;
      }
    }

    length--;
  }

  // nums.sort();

  return (nums.last - 1) * (nums[nums.length - 2] - 1);
}
